### PR TITLE
Sanitise UserProperties schema

### PR DESCRIPTION
**Pull request by t3chguy**
It isn't an event so shouldn't have an eventName

We may wish to further split it into `SetUserProperties` and `SetOnceUserProperties` to provide full context

### CODE REVIEW

**Comment by pixlwave** [comment]
Looks good to me 👍

I made a small PR to your PR to remove the protocol conformance for Swift as this would fail to compile without the event name being present: https://github.com/matrix-org/matrix-analytics-events/pull/29

**Comment by t3chguy** [comment]
@pixlwave in future feel free to push commits onto my branches :)

**Comment by pixlwave** [comment]
Ok thanks, I didn't want to make any assumptions 🙂

**Review by pixlwave — APPROVED** [review]
Works for me on the Swift side of this 👍

File: types/kotlin2/UserProperties.kt [hunk 1]
```diff
@@ -22,14 +22,18 @@ import im.vector.app.features.analytics.itf.VectorAnalyticsEvent
...
+data class UserProperties(
         /**
          * The selected messaging use case during the onboarding flow.
          */
         val ftueUseCaseSelection: FtueUseCaseSelection? = null,
+        /**
+         * Number of spaces (and sub-spaces) the user is joined to
+         */
+        val numSpaces: Int? = null,
 ) : VectorAnalyticsEvent {
```
Comment (ouchadam) [inline types/kotlin2/UserProperties.kt:37]: we'll need to drop extending from this interface as it'll attempt to enforce `getName`

File: stub-generator/matrix_analytics_stub_generator/kotlin.py [hunk 2]
```diff
@@ -26,11 +26,15 @@ def compute_kotlin(schema: Schema) -> str:
 
     if is_screen:
         itf = "VectorAnalyticsScreen"
-    else:
+    elif schema.event_name:
```
Comment (ouchadam) [inline stub-generator/matrix_analytics_stub_generator/kotlin.py:29]: this is my bad, forgot to mention in the other comment, we'll also need to drop the `override` when the itf is blank https://github.com/matrix-org/matrix-analytics-events/pull/28/files#diff-7334dd3531795ac0894703f568db3f5afd7f0410dd5a61abf44f2a78171510a7R60

https://github.com/matrix-org/matrix-analytics-events/pull/28/files#diff-6d7172fb23e30fc3d2fd5fe0c9d311f76085b87e71a19ec37e3f2eba534897a7R100

https://github.com/matrix-org/matrix-analytics-events/pull/28/files#diff-6d7172fb23e30fc3d2fd5fe0c9d311f76085b87e71a19ec37e3f2eba534897a7R102

**Review by ouchadam — APPROVED** [review]

**Pull request merged**